phc/easy/omics/genomic_test.py

```diff
@@ -62,43 +64,54 @@ def transform_results(data_frame: pd.DataFrame, **expand_args):
                 Frame.codeable_like_column_expander("sourceFile"),
             ],
         }
-
-        return (
-            pd.concat(
-                data_frame.apply(
-                    lambda x: pd.DataFrame(
-                        [{"index": x.name, **s} for s in x.sets]
-                    ),
-                    axis=1,
-                ).values
-            )
-            .join(
-                Frame.expand(data_frame, **args).drop(["sets"], axis=1),
-                on="index",
-                rsuffix=".test",
+        df = Frame.expand(data_frame, **args)
+
+        if "sets" in df.columns:
+            df = (
+                pd.concat(
+                    data_frame.apply(
+                        lambda x: pd.DataFrame(
+                            [{"index": x.name, **s} for s in x.sets]
+                        ),
+                        axis=1,
+                    ).values
+                )
+                .join(df.drop(["sets"], axis=1), on="index", rsuffix=".test")
+                .drop(["index"], axis=1)
+                .reset_index(drop=True)
             )
-            .drop(["index"], axis=1)
-            .reset_index(drop=True)
-        )
+
+        test_type = params.get("type", None)
+
+        if test_type and len(df) > 0:
+            # TODO: Remove when API fixed
+
+            # NOTE: The API does not filter the returned sets because it is a
+            # nested structure. Since it's not a boatload of information, we opt
+            # to filter client-side for now.
+            return df[df.setType == test_type].reset_index(drop=True)
+
+        return df
 
     @classmethod
     def get_data_frame(
         cls,
         patient_id: Optional[str] = None,
-        status: Optional[GenomicTestStatus] = None,
+        status: Optional[GenomicTestStatus] = GenomicTestStatus.ACTIVE,
```

Comment:
By default, we'll retrieve the active tests. Remember--the cache will be passed on all the params (and the URL).